remote/rpc/clientV6.go

```diff
@@ -102,6 +103,48 @@ func (p *RPCClientV6) ListServices(ctx context.Context, namespace string) ([]v6.
 	return services, err
 }
 
+func (p *RPCClientV6) ListServicesWithOptions(ctx context.Context, opts v11.ListServicesOptions) ([]v6.ControllerStatus, error) {
+	return listServicesWithOptions(ctx, p, opts, supportedKindsV6)
+}
+
+type listServicesWithOptionsClient interface {
+	ListServices(ctx context.Context, namespace string) ([]v6.ControllerStatus, error)
+}
+
+// listServicesWithOptions polyfills the ListServiceWithOptions()
+// introduced in v11 by removing unwanted resources after fetching
+// all the services.
+func listServicesWithOptions(ctx context.Context, p listServicesWithOptionsClient, opts v11.ListServicesOptions, supportedKinds []string) ([]v6.ControllerStatus, error) {
```

Comment:
This should also check that only one of `opts.Namespace` and `opts.Services` has a value, otherwise you could in principle ask it for specific services, as well a namespace that doesn't include (all of) them.

Reply:
(You could put this cross-version stuff in compat.go)

Reply:
(Oh I see, you're following the precedent set by `listImagesWithOptions`. Carry on.)

Reply:
> Oh I see, you're following the precedent set by listImagesWithOptions. Carry on.)

I actually didn't. I mistakenly thought the `daemon.ListImagesWithOptions()` covers it but it obviously doesn't invoke it.. so will add, thanks!